### PR TITLE
[PLGN-762] Markdown | Snyk

### PR DESCRIPTION
**Pull request by cmcnally-r7**
### Description

Describe the proposed changes:

  - Upgrade PyPandoc dependency from `v1.5` to `v1.13`

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/markdown/help.md [hunk 1]
```diff
@@ -156,34 +155,39 @@ Example input:
...
+*This plugin does not contain any custom output types.*
 
 ## Troubleshooting
 
-_This plugin does not contain any troubleshooting information._
+*There is no troubleshooting for this plugin.*
 
 # Version History
 
+* 3.1.3 - Update PyPandoc dependency
```
Comment (rbowden-r7) [inline plugins/markdown/help.md:190]: might also be worth adding here that the sdk has been bumped

File: plugins/markdown/unit_test/test_markdown_to_txt.py [hunk 2]
```diff
@@ -19,7 +18,7 @@ def setUp(self) -> None:
     @parameterized.expand(
         [
             ({"markdown": "IyBSYXBpZDcgSW5zaWdodENvbm5lY3Q="}, expected_result),
-            ({"markdown_string": "# Rapid7 InsightConnect"}, expected_result),
+            ({"markdown_string": "# Rapid7 InsightConnect", "markdown": ""}, expected_result),
         ]
     )
     def test_markdown_to_txt_valid(self, input_params, expected):
```
Comment (rbowden-r7) [inline plugins/markdown/unit_test/test_markdown_to_txt.py:24]: would it be possible to add in json schema validation to the uni tests ?

**Comment by cmcnally-r7** [comment]
All unit tests are passing locally but 2 fail on the GitHub checks - not sure what's causing this

**Comment by rbowden-r7** [comment]
> All unit tests are passing locally but 2 fail on the GitHub checks - not sure what's causing this

The unit tests failing on Jenkins is likely due to differences in pandoc that's running on you mac vs the pandoc running on ubuntu as the pypandoc package looks to be a wrapper for the os package itself 

```
Pypandoc uses pandoc, so it needs an available installation of pandoc. Pypandoc provides 2 packages, "pypandoc" and "pypandoc_binary", with the second one including pandoc out of the box. The 2 packages are identical, with the only difference being that one includes pandoc, while the other don't.
```

how we install pandoc
```
RUN apt-get update && apt-get install -y pandoc wkhtmltopdf --no-install-recommends
```

**Comment by cmcnally-r7** [comment]
The extra args will cause unit tests to fail on macos but work on ubuntu-bullseye (jenkins).
- ubuntu-bullseye = --atx-headers
- macos = --markdown-headings=atx
(This is due to apt not being able to provide the latest version of pandoc)

**Review by rbowden-r7 — APPROVED** [review]

**Pull request merged**